app/main.py

```
@@ -0,0 +1,330 @@
+        self.thread.started.connect(self.worker.run)
+        self.worker.finished.connect(self.on_compression_finished)
+        self.worker.error.connect(self.on_compression_error)
+        self.worker.log_message.connect(self.log_edit.append)
+        self.worker.progress_updated.connect(self.update_progress)
+        
+        self.thread.start()
+
+    def stop_compression(self):
+        self.status_bar.showMessage("Stopping...")
+        if self.worker:
+            self.worker.stop() # Tell worker to stop gracefully
```

Flake8 (7.2.0)

[error] at least two spaces before inline comment

(E261)